src/runtime/composables/authjs/useAuth.ts

```diff
@@ -112,8 +112,12 @@ const signIn: SignInFunc<SupportedProviders, SignInResult> = async (provider, op
 
   const csrfToken = await callWithNuxt(nuxt, getCsrfToken)
 
-  const headers: { 'Content-Type': string, 'cookie'?: string | undefined } = {
+  const headers: { 'Content-Type': string, 'X-Auth-Return-Redirect': string, 'cookie'?: string | undefined } = {
     'Content-Type': 'application/x-www-form-urlencoded',
+    // This header is set to force AuthJS to return JSON
+    // https://github.com/nextauthjs/next-auth/blob/b86f7bb721e2fa4e9c3251b029095b0f50d46198/packages/next-auth/src/react.tsx#L270-L285
+    // https://github.com/nextauthjs/next-auth/blob/b86f7bb721e2fa4e9c3251b029095b0f50d46198/packages/core/src/index.ts#L160
+    'X-Auth-Return-Redirect': '1',
```

Comment:
This is a new header which needs to be documented (or linked to `authjs` docs). We need to mention for users to add it to their header whitelists